Cleanup BitArray tests

hughbe opened the pull request.

hughbe commented:
Test Innerloop OSX Debug Build and Test

stephentoub commented on src/System.Collections/tests/BitArray/BitArray_CtorTests.cs:
```diff
@@ -2,84 +2,84 @@
+                Assert.False(bitArray[i]);
+                Assert.False(bitArray.Get(i));
+            }
+            Ctor_BitArray(bitArray);
```
Clever

stephentoub commented on src/System.Collections/tests/BitArray/BitArray_OperatorsTests.cs:
```diff
@@ -2,242 +2,177 @@
-                        Assert.False(ba4.Get(i)); //"Err_6! ba4.Get(" + i + ") should be false"
-                    else
-                        Assert.True(ba4.Get(i)); //"Err_7! ba4.Get(" + i + ") should be true"
+                    Assert.False(bitArrayNot[i]);
```
Why are the if/else blocks the same?

hughbe replied:
Good catch - fixed

stephentoub commented:
Test Innerloop OSX Debug Build and Test please

hughbe commented:
@stephentoub I obviously ran a coverage report for System.Collections after cleaning up the tests. I expect a slight increase in coverage too. However, BitArray doesn't come up in the coverage report for some reason - it doesn't come up on the http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/code_coverage_windows/Code_Coverage_Report/ either

stephentoub commented:
> However, BitArray doesn't come up in the coverage report for some reason

It doesn't come up in the coverage report because BitArray is coming from mscorlib; the implementation in corefx is only being compiled in for corert:
https://github.com/dotnet/corefx/blob/master/src/System.Collections/src/System.Collections.csproj#L49-L50
so you'll need to do a coverage run including mscorlib in order to see BitArray coverage.

@mellinoe, is this by design?  Is there a reason we can't use BitArray from corefx?

ianhays commented:
LGTM.

>Is there a reason we can't use BitArray from corefx?

For coverage generation specifically, or for the build in general? I didn't think the latter was possible  since it's already exposed in mscorlib. The former isn't possible to my knowledge without the latter also being true i.e. to get coverage for BitArray using stock OpenCover we need to run the tests against the BitArray in CoreFX or we need to go through the trouble of running against a locally built mscorlib with accompanying pdb.

stephentoub commented:
> for the build in general?

For build, i.e. code using BitArray would use one compiled into System.Collections.NonGeneric.dll rather than System.Collections.NonGeneric.dll type forwarding to the one in mscorlib.

> I didn't think the latter was possible since it's already exposed in mscorlib

Are there APIs exposed from mscorlib, other than BitArray itself, that use BitArray in a public signature?

ianhays commented:
>Are there APIs exposed from mscorlib, other than BitArray itself, that use BitArray in a public signature?

Thanks for clarifying. I don't know of a good way to check this. Maybe I could remove BitArray from the exports list in mscorlib and then try building to see if I get errors from another class?

stephentoub commented:
@hughbe, I checked the coverage of BitArray before and after your changes, and it looks like the branch coverage actually dropped.  Can you see why?

hughbe commented:
@stephentoub thanks for running coverage reports - I'm getting some sort of infinite loop while running the tests, I think it may be related to the latest pdb/coverage changed in Buildtools. I'll try deleting and recloning corefx but this will take time - do you have a copy of the new coverage report on hand - perhaps you could link it on this PR if you have it (don't worry if you don't)

stephentoub commented:
> do you have a copy of the new coverage report on hand

I don't, sorry.

hughbe commented:
That's okay, I'll try to fix the coverage drop within the next day or so - sorr

hughbe commented:
Test Innerloop Windows_NT Debug Build and Test

hughbe commented:
Thanks @stephentoub for running a coverage report. I did some investigation and fixed the drop in coverage - I accidentally removed some test code that decreases then increases the size of the BitArray
At the same time, I added some more tests (for ICollection methods and properties, Ctor_IntArray and Ctor_ByteArray

stephentoub commented:
LGTM
